compiler/src/dotty/tools/dotc/transform/SelectStatic.scala

```diff
@@ -29,13 +29,23 @@ class SelectStatic extends MiniPhase with IdentityDenotTransformer {
       sym.isScalaStatic
     val isStaticRef = !sym.is(Package) && !sym.maybeOwner.is(Package) && isStaticMember
     val tree1 =
-      if (isStaticRef && !tree.qualifier.symbol.isAllOf(JavaModule) && !tree.qualifier.isType)
-        Block(List(tree.qualifier), ref(sym))
+      if isStaticRef && !tree.qualifier.symbol.isAllOf(JavaModule) && !tree.qualifier.isType then
+        if isStaticOwnerRef(tree.qualifier) then ref(sym)
```

Comment:
Since this is quite subtle, a comment would be welcome (either here or in the doc comment of the class)

Reply:
I've significantly expanded the doc comment of the class.